activation/wire/wire_v2.go

```diff
@@ -0,0 +1,76 @@
+package wire
+
+import "github.com/spacemeshos/go-spacemesh/common/types"
+
+//go:generate scalegen
+
+type ActivationTxV2 struct {
+	InnerActivationTxV2
+
+	SmesherID types.NodeID
+	// Signature over InnerActivationTxV2
+	Signature types.EdSignature
+}
+
+type InitialAtxPartsV2 struct {
+	CommitmentATX types.Hash32
```

Comment:
We definitely want to avoid having the CommitmentATX twice in the ATX. I'd also like to avoid having too many object / types that form the ATX - what if we just inline this type and make each field separately optional?

Reply:
It's not twice in the ATX.

> what if we just inline this type and make each field separately optional?

The whole point of creating this sub-struct is that it's only possible to either set all of them (in an initial ATX) or none (any other ATX) at once. It simplifies verification by limiting the number of possible combinations and making some of the invalid combinations impossible to create (for example it's not possible to have `CommitmentATX == nil && InitialPost != nil`.